crates/topos-tce/src/app_context.rs

```diff
@@ -141,11 +143,35 @@ impl AppContext {
 
     pub async fn shutdown(&mut self) -> Result<(), Box<dyn std::error::Error>> {
         info!("Shutting down the TCE client...");
+
         self.api_client.shutdown().await?;
         self.tce_cli.shutdown().await?;
         self.gatekeeper.shutdown().await?;
         self.network_client.shutdown().await?;
```

Comment:
Maybe we should crash here too if they fail?

Reply:
Also: I think it would be good to have a timeout in this method, forcing shutdown after `n` seconds regardless of any subsystem being slow to shutdown. Daemons that don't shutdown properly are so annoying.

Reply:
Mmmh there is two approach here, either we define that we have a timeout that kill everything internally, or we let the system management using a SIGKILL after X sec.
On a larger scope, I'm not really satisfied with the current handling of the shutdown, we should be safe to cut the service without cleanup at any time and being able to recover on startup. It may need more work on various part tho.

Reply:
> we should be safe to cut the service without cleanup at any time and being able to recover on startup

Agreed. I think it's the kind of discipline that has benefits beyond just shutdowns. It kind of forces us to write correct code.